modules/widgets/simple-payments/customizer.js

```diff
@@ -70,6 +73,31 @@
 		widgetForm.find( '.jetpack-simple-payments-delete-product' ).on( 'click', deleteProduct( widgetForm ) );
 	}
 
+	function syncProductLists() {
+		var request = wp.ajax.post( 'customize-jetpack-simple-payments-buttons-get', {
+			'customize-jetpack-simple-payments-nonce': api.settings.nonce[ 'customize-jetpack-simple-payments' ],
+			'customize_changeset_uuid': api.settings.changeset.uuid
+		} );
+
+		request.done( function( data ) {
+			var selectedProduct = 0;
+
+			$( document ).find( 'select.jetpack-simple-payments-products' ).each( function( index, select ) {
+				var $select = $( select );
+				selectedProduct = $select.val();
+
+				$select
+					.find( 'option' )
+					.remove()
+					.end()
+					.append( $.map( data, function( product ) {
+						return $( '<option>', { value: product.ID, text: product.post_title } );
+					} ) )
+					.val( selectedProduct );
```

Comment:
what does this line do? could we add a comment at the end tersely explaining what `selectedProduct` is here for and what its value will be?

Reply:
The logic there is:

- temporarily store the `<select>` selected product
- empty the `<select>`
- fill the `<select>` with the updated products
- re-select the product that was selected before the switcheroo

I'm fine with over-documenting stuff, but in this particular case I'd say it's just a matter of remembering the good old days of yore when we were more used to look at jQuery code. 😄

Reply:
I guess I'm getting lost even with the jQuery calls

the `end()` in the sequence is awkward since it's _not at the end_
the last `.val( selectedProduct )` is also a bit puzzling to me since it gives me the impression that it's resetting the value of the `select` to the value polled in line 87R.

even quick explanation of what's being set to what in line 96R would go a long way.

Reply:
jQuery, known for it's awkwardness defines `end()` as:

>End the most recent filtering operation in the current chain and return the set of matched elements to its previous state.

In this particular case, it _ends_ the `find` filter operation by returning `$select`, where we can append the updated list of `option` elements and restore the selected value.

Reply:
I won't say anymore after this 🤝 but I think that the fact that we have to refer to documentation and answer this here is indication that someone in the future is going to come in and have no idea what's going on when a single sentence or fragment could lead them in on it…

```js
…
	.end() // end current operation and return original element
	…
	.val( selectedProduct ); // and append into the original `<select>`
```

I clearly don't understand what I'm talking about, but I question whether the next person who comes around to this will either

Reply:
I've added a comment for the `end` call on 719071c9d9e5e62b0e6aa222ca02bd65be750518.